airflow/dag_processing/manager.py

```diff
@@ -959,7 +973,15 @@ def set_file_paths(self, new_file_paths):
         :return: None
         """
         self._file_paths = new_file_paths
+
+        # clean up the queues; remove anything queued which no longer in the list, including callbacks
         self._file_path_queue = collections.deque(x for x in self._file_path_queue if x in new_file_paths)
+        Stats.gauge("dag_processing.file_path_queue_size", len(self._file_path_queue))
+
+        callback_paths_to_del = list(x for x in self._callback_to_execute.keys() if x not in new_file_paths)
```

Comment:
This just closes a small mem-leak; we removed the files from the queue, but we didn't remove their callbacks.